compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -985,23 +1033,46 @@ options numProcessors =
         <> cmdClean
       )
 
-parserInfo :: Int -> ParserInfo Command
+parserInfo :: Int -> Opt.ParserInfo Command
 parserInfo numProcessors =
-  info (helper <*> options numProcessors)
-    (  fullDesc
-    <> progDesc "Invoke the DAML compiler. Use -h for help."
-    <> headerDoc (Just $ PP.vcat
+  Opt.info (Opt.helper <*> options numProcessors)
+    (  Opt.fullDesc
+    <> Opt.progDesc "Invoke the DAML compiler. Use -h for help."
+    <> Opt.headerDoc (Just $ PP.vcat
         [ "damlc - Compiler and IDE backend for the Digital Asset Modelling Language"
         , buildInfo
         ])
     )
 
+cliArgsFromDamlYaml :: IO [String]
+cliArgsFromDamlYaml = do
+    handle (\(_ :: ConfigError) -> return [])
+           $ do
+               project <- readProjectConfig $ ProjectPath "."
+               case parseProjectConfig project of
+                   Left _ -> return []
+                   Right pkgConfig -> case cliOpts pkgConfig of
+                       Nothing -> return []
+                       Just xs -> return xs
+
 main :: IO ()
 main = do
     -- We need this to ensure that logs are flushed on SIGTERM.
     installSignalHandlers
     numProcessors <- getNumProcessors
-    withProgName "damlc" $ join $ execParserLax (parserInfo numProcessors)
+    let parse = ParseArgs.lax (parserInfo numProcessors)
+    cliArgs <- Env.getArgs
+    damlYamlArgs <- cliArgsFromDamlYaml
+    let (_, tempParseResult) = parse cliArgs
+    Command(cmd, _) <- Opt.handleParseResult tempParseResult
+    let args = if cmd `Set.member` Set.fromList [Build, Compile, Ide]
+               then cliArgs ++ damlYamlArgs
+               else cliArgs
+        (errMsgs, parseResult) = parse args
+    Command(_, io) <- Opt.handleParseResult parseResult
```

Comment:
Parsing twice looks a bit ugly but I don’t have a better suggetion (if we limit it to `ghc-options` we could do better). Probably worth adding a comment that explains why you parse twice.